src/flag_gems/testing/__init__.py

```diff
@@ -0,0 +1,21 @@
+import torch
+
+RESOLUTION = {
+    torch.bool: 0,
+    torch.int16: 0,
+    torch.int32: 0,
+    torch.float16: 1e-3,
+    torch.float32: 1.3e-6,
+    torch.bfloat16: 0.016,
+}
+
+
+def assert_close(res, ref, dtype, equal_nan=False, reduce_dim=1):
+    ref = ref.to(dtype)
+    atol = 1e-4 * reduce_dim
+    rtol = RESOLUTION[dtype]
+    torch.testing.assert_close(res, ref, atol=atol, rtol=rtol, equal_nan=equal_nan)
```

Comment:
Two issues:
- fp64 is not in RESOLUTION's key set.
- passing a dtype matching neither ref nor res may raise an error.

Reply:
Also, do we consider default dtype=None? Now dtype is mandate.

Reply:
> Two issues:
> 
> * fp64 is not in RESOLUTION's key set.
> * passing a dtype matching neither ref nor res may raise an error.

Now only isclose & allclose test fp64. These two tests do not use assert_close.
I will assert res.dtype == dtype here.

Reply:
> > Two issues:
> > 
> > * fp64 is not in RESOLUTION's key set.
> > * passing a dtype matching neither ref nor res may raise an error.
> 
> Now only isclose & allclose test fp64. These two tests do not use assert_close. Maybe we can assert res.dtype == dtype here?

done